hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerStateManager.java

```diff
@@ -296,10 +296,10 @@ ContainerInfo allocateContainer(
         .setReplicationFactor(pipeline.getFactor())
         .setReplicationType(pipeline.getType())
         .build();
-    pipelineManager.addContainerToPipeline(pipeline.getId(),
```

Comment:
I see that this method (allocateContainer) is also called from places where lock on the pipeline is not held.

Reply:
Should we also add a comment that this method must be called with lock held on the pipeline?

Reply:
Addressed in the latest patch.